extensions/amp-instagram/1.0/test/test-instagram.js

```diff
@@ -0,0 +1,134 @@
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../../src/preact';
+import {Instagram, handleMessage} from '../instagram';
+import {mount} from 'enzyme';
+import {useRef} from '../../../../src/preact';
+
+describes.sandboxed('Instagram preact component v1.0', {}, (env) => {
+  it('Normal render', () => {
+    const el = document.createElement('div');
+    document.body.appendChild(el);
+    const wrapper = mount(
+      <Instagram
+        shortcode="B8QaZW4AQY_"
+        style={{
+          'width': 500,
+          'height': 600,
+        }}
+      />,
+      {attachTo: el}
+    );
+
+    const iframe = wrapper.find('iframe');
+
+    expect(iframe.prop('src')).to.equal(
+      'https://www.instagram.com/p/B8QaZW4AQY_/embed/?cr=1&v=12'
+    );
+    expect(wrapper.find('iframe').prop('style').width).to.equal('100%');
+    expect(wrapper.find('iframe').prop('style').height).to.equal('100%');
+    expect(wrapper.find('div')).to.have.lengthOf(2);
+  });
+
+  it('Render with caption', () => {
+    const el = document.createElement('div');
+    document.body.appendChild(el);
+    const wrapper = mount(
+      <Instagram
+        shortcode="B8QaZW4AQY_"
+        captioned
+        style={{'width': 500, 'height': 705}}
+      />,
+      {attachTo: el}
+    );
+    expect(wrapper.find('iframe').prop('src')).to.equal(
+      'https://www.instagram.com/p/B8QaZW4AQY_/embed/captioned/?cr=1&v=12'
+    );
+    expect(wrapper.find('iframe').prop('style').width).to.equal('100%');
+    expect(wrapper.find('iframe').prop('style').height).to.equal('100%');
+    expect(wrapper.find('div')).to.have.lengthOf(2);
+  });
+
+  it('Resize prop is called', () => {
+    mount(
+      <Instagram
+        shortcode="B8QaZW4AQY_"
+        style={{'width': 500, 'height': 600}}
+        requestResize={env.sandbox.spy()}
+      />,
+      {attachTo: document.body}
+    );
+
+    const requestResizeSpy = env.sandbox.spy();
+    const heightSpy = env.sandbox.spy();
+    const opacitySpy = env.sandbox.spy();
+
+    handleMessage(
+      createMockEvent(),
+      useRef({'contentWindow': 'source'}),
+      requestResizeSpy,
+      heightSpy,
+      opacitySpy
+    );
+
+    expect(requestResizeSpy).to.have.been.calledOnce;
+    expect(heightSpy).to.have.not.been.called;
+    expect(opacitySpy).to.have.been.calledOnce;
+  });
+
+  it('Height is changed', () => {
+    mount(
+      <Instagram
+        shortcode="B8QaZW4AQY_"
+        style={{'width': 500, 'height': 600}}
+        requestResize={env.sandbox.spy()}
+      />,
+      {attachTo: document.body}
+    );
+
+    const heightSpy = env.sandbox.spy();
+    const opacitySpy = env.sandbox.spy();
+
+    handleMessage(
+      createMockEvent(),
+      useRef({'contentWindow': 'source'}),
```

Comment:
I don't quite understand this. A `contentWindow` is an object. Further, you already have a pretty good and usable `contentWindow` for the iframe that you rendered above, why not just use that?

Reply:
I'm setting `contentWindow` as a manual ref (`'source'`) in order to test the `handleMessage` function that's called after the layout. That way, the function can be tested without connecting to the Instagram server in case of any changes to the mounted `contentWindow`. I was also having issues properly grabbing a ref of the iframe using the mounted iframe above, as there was no ref assigned to the HTML that was rendered. As a result, it was hard for me to grab attributes like `contentWindow` and I decided it would be better to use a mock ref instead. Lastly, I actually think the mounting of the iframe is unnecessary here since we are only testing a function that's called within it, so we can just test the `handleMessage` function directly. What do you think?